File: MPP/Lab7_2/Network/src/main/java/Proxy/DestinatieProxy.java
package Proxy;

import ProjectModels.Destinatie;
import ServiceInterface.IServiceDestinatie;
import Utils.Request;
import Utils.Response;

import java.io.IOException;
import java.io.ObjectInputStream;
import java.io.ObjectOutputStream;
import java.net.Socket;
import java.time.LocalDateTime;
import java.util.List;

public class DestinatieProxy implements IServiceDestinatie {
    private ObjectInputStream input;
    private ObjectOutputStream output;
    private Socket connection;
    private ReadResponse r;

    public DestinatieProxy(ObjectInputStream input, ObjectOutputStream output, Socket connection, ReadResponse r) {
        this.input = input;
        this.output = output;
        this.connection = connection;
        this.r = r;
    }

    @Override
    public int getIdDestinatie(String destinatie, LocalDateTime data) {
        Request req = new Request("getIdDestinatie");
        System.out.println(data.toString());
        req.setDestinatieStr(destinatie);
        req.setLocalDateTime(data.toString());
        try{
            output.writeObject(req);
            output.flush();
            Response resp = r.readResponse();
            if(resp.getMesaj().equals("ok"))
            {
                return resp.getId();
            }
        } catch (IOException e) {
            e.printStackTrace();
        }

        return 0;
    }

    @Override
    public List<Destinatie> returnLista() {
        Request cer = new Request("getListDestinatii");
        try{
            output.writeObject(cer);
            output.flush();
            Response re = r.readResponse();
            if(re.getMesaj().equals("ok"))
            {
                return re.getListDestinatii();
            }
        } catch (IOException e) {
            e.printStackTrace();
        }

        return null;
    }

    @Override
    public Destinatie getDestinatie(int id) {
        Request c = new Request("getDestinatie");
        c.setId(id);
        try{
            output.writeObject(c);
            output.flush();
            Response res = r.readResponse();
            if(res.getMesaj().equals("ok"))
            {
                return res.getDestinatie();
            }
        } catch (IOException e) {
            e.printStackTrace();
        }
        return null;
    }

    @Override
    public void update(Destinatie dest) {
        Request request = new Request("Update");
        request.setDestinatie(dest);
        try{
            output.writeObject(request);
            output.flush();
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

File: MPP/ServerC#CrossPlatform/ConsoleApplication1/Networking/DestinatieProxy.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Net.Sockets;
using System.Runtime.Serialization.Formatters.Binary;
using System.Text;
using System.Threading.Tasks;
using Model;
using Service;

namespace Networking
{
   public  class DestinatieProxy:IServiceDestinatii
    {
        NetworkStream stream;
        BinaryFormatter formatter;
        TcpClient connection;
        ReadRespone r;

        public DestinatieProxy(NetworkStream s ,TcpClient c  , ReadRespone rr)
        {
            stream = s;
            connection = c;
            r = rr;
            formatter = new BinaryFormatter();
        }

        public DataSet DestinatiiRep()
        {
            Request req = new Request("DestinatiiRep");
            formatter.Serialize(stream, req);
            stream.Flush();
            Response re = r.readResponse();
            if(re.getMesaj().Equals("ok"))
            {
                return re.getDs();
            }
            return null;
        }

        public int getIdDestinatie(string destinatie, DateTime timp)
        {
            Request req = new Request("getIdDestinatie");
            req.setLocalDateTime(timp.ToString());
            req.setDestinatieStr(destinatie);
            formatter.Serialize(stream, req);
            stream.Flush();
            Response re = r.readResponse();
            if (re.getMesaj().Equals("ok"))
            {
                return re.getId();
            }
            return -1;

        }

        public Destinatie getDestinatie(int idDest)
        {
            Request req = new Request("getDestinatie");
            req.setId(idDest);
            formatter.Serialize(stream, req);
            stream.Flush();
            Response resp = r.readResponse();
            if(resp.getMesaj().Equals("ok"))
            {
                return resp.getDestinatie();
            }
            return null;

        }

        public void update(Destinatie dest)
        {
            Request req = new Request("update");
            req.setDestinatie(dest);
            formatter.Serialize(stream, req);
            stream.Flush();
        }
    }
}
